AX: Make AXCoreObject ref-counting non-threadsafe, improving performance

twilco opened the pull request:
#### 139e9a6d68c9f8638cee836da5d04690f8a9d323
<pre>
AX: Make AXCoreObject ref-counting non-threadsafe, improving performance
<a href="https://bugs.webkit.org/show_bug.cgi?id=293164">https://bugs.webkit.org/show_bug.cgi?id=293164</a>
<a href="https://rdar.apple.com/151499521">rdar://151499521</a>

Reviewed by Joshua Hoffman.

AXCoreObject ref-counting had to be threadsafe because prior to this commit, AXIsolatedObjects were built on the
main-thread (incurring numerous refs along the way), and then &quot;transferred&quot; to the accessibility thread. However, there
were other places we held refs or otherwise manipulated isolated objects on the main-thread, even after transferring them:

  - Attaching them to wrappers
  - Returning the root isolated tree node from the WKAccessibilityWebPageObjectBase
  - Some usages of associatedAXObject() and performFunctionOnMainThread (which is async) implicitly assumed the associated
    isolated object would stay alive for the lifetime of associatedAXObject() / the dispatched function, which can no
    longer be relied upon.

With this commit, instead of creating isolated objects on the main-thread, we instead create the data needed to populate
an isolated object, and send that plain-struct over. Then the accessibility thread can create the isolated object, and
thus be the only thing that ever refs and derefs these objects, in turn allowing us to safely make our ref-counting
non-threadsafe.

This required a host of associated changes:

  - Rather than storing the isolated tree root node on WKAccessibilityWebPageObjectBase, we instead store a thread-safe
    pointer to the isolated tree. AXIsolatedTree is still threadsafe ref-counted, so it&apos;s OK to use from either thread,
    and we can use it to get the root node when we know we&apos;re off the main-thread.

  - The AttachWrapper enum is removed. All isolated objects are now unconditionally attached on the accessibility thread.

  - WebAccessibilityObjectWrapperBase::m_isolatedObjectInitialized has become an atomic bool, as the secondary thread
    writes it when attaching an isolated object to a wrapper, and the main-thread reads it when destroying a main-thread
    object to know whether it should call NSAccessibilityUnregisterUniqueIdForUIElement. This probably should&apos;ve always
    been atomic, as it was possible to attach isolated objects to wrappers off the main-thread before this commit too.

  - Because all wrapper attachments happen off the main-thread now, there&apos;s nothing to kick off the first AXIsolatedTree::applyPendingChanges.
    Usually that happens when we get a backing object in the wrapper, and call updateObjectBackingStore. But until the
    first applyPendingChanges, there is no backingObject, so it&apos;s a classic chicken and egg problem. Solve this by adding
    AXTreeStore::applyPendingChangesForAllIsolatedTrees, and calling it whenever a wrapper is missing a backing object,
    as it may be there, just waiting to be attached.

This change has huge performance wins. On pages with slow VO-Right navigation (e.g. <a href="http://html.spec.whatwg.org)">http://html.spec.whatwg.org)</a>,
ref-counting took up ~40% of the samples, largely because threadsafe ref-counting is slow. It also means iOS no longer
unnecessarily pays the cost of threadsafe ref-counting that it didn&apos;t even use, since isolated tree mode is not enabled
for that platform.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Add AXTreeStore.cpp.
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::headingLevel const):
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::AXCoreObject):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::setIsolatedTree):
(WebCore::AXObjectCache::rootObjectForFrame):
(WebCore::AXObjectCache::isolatedTreeRootObject): Deleted.
(WebCore::AXObjectCache::setIsolatedTreeRoot): Deleted.
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXSearchManager.cpp:
(WebCore::appendChildrenToArray):
* Source/WebCore/accessibility/AXTreeStore.h:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::headingTagLevel const): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::insertChild):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::scrollView const):
* Source/WebCore/accessibility/atspi/AXObjectCacheAtspi.cpp:
(WebCore::AXObjectCache::platformPerformDeferredCacheUpdate):
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
(WebCore::AXIsolatedObject::AXIsolatedObject):
(WebCore::AXIsolatedObject::create):
(WebCore::isDefaultValue):
(WebCore::AXIsolatedObject::associatedAXObject const):
(WebCore::AXIsolatedObject::setMathscripts):
(WebCore::AXIsolatedObject::setObjectProperty):
(WebCore::AXIsolatedObject::setObjectVectorProperty):
(WebCore::AXIsolatedObject::setProperty):
(WebCore::AXIsolatedObject::insertText):
(WebCore::AXIsolatedObject::press):
(WebCore::AXIsolatedObject::initializeProperties): Deleted.
(WebCore::AXIsolatedObject::canBeMultilineTextField): Deleted.
(WebCore::AXIsolatedObject::setOptionalProperty): Deleted.
(WebCore::AXIsolatedObject::scrollView const): Deleted.
(WebCore::AXIsolatedObject::headingTagLevel const): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::AXIsolatedTree):
(WebCore::AXIsolatedTree::createEmptyContent):
(WebCore::AXIsolatedTree::create):
(WebCore::AXIsolatedTree::applyPendingRootNodeLocked):
(WebCore::AXIsolatedTree::storeTree):
(WebCore::AXIsolatedTree::nodeChangeForObject):
(WebCore::AXIsolatedTree::queueChange):
(WebCore::AXIsolatedTree::addUnconnectedNode):
(WebCore::AXIsolatedTree::resolveAppends):
(WebCore::AXIsolatedTree::queueAppendsAndRemovals):
(WebCore::AXIsolatedTree::collectNodeChangesForSubtree):
(WebCore::AXIsolatedTree::updateNode):
(WebCore::AXIsolatedTree::updateChildren):
(WebCore::AXIsolatedTree::setPendingRootNodeID):
(WebCore::AXIsolatedTree::setPendingRootNodeIDLocked):
(WebCore::AXIsolatedTree::willBeDestroyed):
(WebCore::AXIsolatedTree::applyPendingChanges):
(WebCore::AXIsolatedTree::processQueuedNodeUpdates):
(WebCore::convertToPropertyFlag):
(WebCore::setPropertyIn):
(WebCore::shouldCacheElementName):
(WebCore::canBeMultilineTextField):
(WebCore::createIsolatedObjectData):
(WebCore::AXIsolatedTree::applyPendingRootNode): Deleted.
(WebCore::AXIsolatedTree::setPendingRootNodeLocked): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.h:
(WebCore::IsolatedObjectData::IsolatedObjectData):
(WebCore::IsolatedObjectData::setProperty):
(WebCore::AXIsolatedTree::axObjectCacheID const):
(WebCore::AXIsolatedTree::rootNode):
(WebCore::AXIsolatedTree::NodeChange::NodeChange):
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
(WebCore::appendBasePlatformProperties):
(WebCore::appendPlatformProperties):
(WebCore::AXIsolatedObject::initializeBasePlatformProperties): Deleted.
(WebCore::AXIsolatedObject::initializePlatformProperties): Deleted.
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase attachIsolatedObject:]):
(-[WebAccessibilityObjectWrapperBase hasIsolatedObject]):
(-[WebAccessibilityObjectWrapperBase detachIsolatedObject:]):
(-[WebAccessibilityObjectWrapperBase updateObjectBackingStore]):
(-[WebAccessibilityObjectWrapperBase axBackingObject]):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper _accessibilityShowContextMenu]):
(-[WebAccessibilityObjectWrapper _accessibilityChildrenFromIndex:maxCount:returnPlatformElements:]):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyFrameLoaderClient::setIsolatedTree):
(WebCore::EmptyFrameLoaderClient::setAXIsolatedTreeRoot): Deleted.
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::setIsolatedTree):
(WebKit::WebLocalFrameLoaderClient::setAXIsolatedTreeRoot): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectBase.h:
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectBase.mm:
(-[WKAccessibilityWebPageObjectBase accessibilityRootObjectWrapper:]):
(-[WKAccessibilityWebPageObjectBase setIsolatedTree:]):
(-[WKAccessibilityWebPageObjectBase setIsolatedTreeRoot:]): Deleted.
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::setIsolatedTree):
(WebKit::WebPage::setAXIsolatedTreeRoot): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.h:

Canonical link: <a href="https://commits.webkit.org/295174@main">https://commits.webkit.org/295174@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc4630dad7701da345b20294d5656222a9a38c48

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104182 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23886 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14207 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109380 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54850 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106222 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32430 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79128 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107188 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18843 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93995 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59455 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18651 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12033 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54210 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88358 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12091 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111764 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31338 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23103 "Found 2 new test failures: http/tests/navigation/ping-attribute/cross-site-url-anchor-cookie-preexisting-cookie.html http/tests/navigation/ping-attribute/cross-site-url-anchor-cookie.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88149 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31702 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90182 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87808 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22379 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32702 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10457 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25802 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31267 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36580 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31061 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34397 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32621 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

AndresGonzalezApple commented on Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
```diff
@@ -52,10 +52,11 @@ struct AXTextRuns;
     ~AXIsolatedObject();
 
     std::optional<AXID> treeID() const final { return tree()->treeID(); }
+    AXID axObjectCacheID() const { return tree()->axObjectCacheID(); }
```
How can axObjectCacheID() be different from treeID()? There should be only one tree ID associated with an object regardless whether it is a live or an isolated object.
Why treeID() returns an optional? I don't think we can have objects that don't belong to a tree.

twilco replied:
> How can axObjectCacheID() be different from treeID()? There should be only one tree ID associated with an object regardless whether it is a live or an isolated object.

Great point, thanks for catching this! I've removed this method in the latest commit.

> Why treeID() returns an optional? I don't think we can have objects that don't belong to a tree.

You're right, it shouldn't return an optional. I think this was changed by someone else a while ago. I added a FIXME for us to handle in a follow-up patch.

minorninth commented on Source/WebCore/accessibility/AXSearchManager.cpp:
```diff
@@ -242,6 +242,7 @@ static void appendChildrenToArray(Ref<AXCoreObject> object, bool isForward, RefP
         // We should only ever hit this case with a live object (not an isolated object), as it would require startObject to be ignored,
         // and we should never have created an isolated object from an ignored live object.
         // FIXME: This is not true for ENABLE(INCLUDE_IGNORED_IN_CORE_AX_TREE), fix this before shipping it.
+        // FIXME: We hit this ASSERT on google.com.
```
Should we at least understand what's going on here before merging this?

twilco replied:
Sure, can investigate! Shouldn't be too hard to figure out what's going on here.

twilco replied:
I added some logging, and confirmed this branch does result in different behavior. All of this code (including the ASSERT) was written at a time when isolated objects could not be ignored, hence this assert. But that assumption is now wrong with `ENABLE(INCLUDE_IGNORED_IN_CORE_AX_TREE)`, so we need to fix this, though the fact that we don't handle this right doesn't cause any readily apparent problems on google.com. I did file a bug to track this so we can address in a follow-up.

twilco committed:
AX: Make AXCoreObject ref-counting non-threadsafe, improving performance
https://bugs.webkit.org/show_bug.cgi?id=293164
rdar://151499521

Reviewed by Joshua Hoffman.

AXCoreObject ref-counting had to be threadsafe because prior to this commit, AXIsolatedObjects were built on the
main-thread (incurring numerous refs along the way), and then "transferred" to the accessibility thread. However, there
were other places we held refs or otherwise manipulated isolated objects on the main-thread, even after transferring them:

  - Attaching them to wrappers
  - Returning the root isolated tree node from the WKAccessibilityWebPageObjectBase
  - Some usages of associatedAXObject() and performFunctionOnMainThread (which is async) implicitly assumed the associated
    isolated object would stay alive for the lifetime of associatedAXObject() / the dispatched function, which can no
    longer be relied upon.

With this commit, instead of creating isolated objects on the main-thread, we instead create the data needed to populate
an isolated object, and send that plain-struct over. Then the accessibility thread can create the isolated object, and
thus be the only thing that ever refs and derefs these objects, in turn allowing us to safely make our ref-counting
non-threadsafe.

This required a host of associated changes:

  - Rather than storing the isolated tree root node on WKAccessibilityWebPageObjectBase, we instead store a thread-safe
    pointer to the isolated tree. AXIsolatedTree is still threadsafe ref-counted, so it's OK to use from either thread,
    and we can use it to get the root node when we know we're off the main-thread.

  - The AttachWrapper enum is removed. All isolated objects are now unconditionally attached on the accessibility thread.

  - WebAccessibilityObjectWrapperBase::m_isolatedObjectInitialized has become an atomic bool, as the secondary thread
    writes it when attaching an isolated object to a wrapper, and the main-thread reads it when destroying a main-thread
    object to know whether it should call NSAccessibilityUnregisterUniqueIdForUIElement. This probably should've always
    been atomic, as it was possible to attach isolated objects to wrappers off the main-thread before this commit too.

  - Because all wrapper attachments happen off the main-thread now, there's nothing to kick off the first AXIsolatedTree::applyPendingChanges.
    Usually that happens when we get a backing object in the wrapper, and call updateObjectBackingStore. But until the
    first applyPendingChanges, there is no backingObject, so it's a classic chicken and egg problem. Solve this by adding
    AXTreeStore::applyPendingChangesForAllIsolatedTrees, and calling it whenever a wrapper is missing a backing object,
    as it may be there, just waiting to be attached.

This change has huge performance wins. On pages with slow VO-Right navigation (e.g. http://html.spec.whatwg.org),
ref-counting took up ~40% of the samples, largely because threadsafe ref-counting is slow. It also means iOS no longer
unnecessarily pays the cost of threadsafe ref-counting that it didn't even use, since isolated tree mode is not enabled
for that platform.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Add AXTreeStore.cpp.
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::headingLevel const):
* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::AXCoreObject):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::setIsolatedTree):
(WebCore::AXObjectCache::rootObjectForFrame):
(WebCore::AXObjectCache::isolatedTreeRootObject): Deleted.
(WebCore::AXObjectCache::setIsolatedTreeRoot): Deleted.
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AXSearchManager.cpp:
(WebCore::appendChildrenToArray):
* Source/WebCore/accessibility/AXTreeStore.h:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::AccessibilityNodeObject::headingTagLevel const): Deleted.
* Source/WebCore/accessibility/AccessibilityNodeObject.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::insertChild):
* Source/WebCore/accessibility/AccessibilityObject.h:
(WebCore::AccessibilityObject::scrollView const):
* Source/WebCore/accessibility/atspi/AXObjectCacheAtspi.cpp:
(WebCore::AXObjectCache::platformPerformDeferredCacheUpdate):
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
(WebCore::AXIsolatedObject::AXIsolatedObject):
(WebCore::AXIsolatedObject::create):
(WebCore::isDefaultValue):
(WebCore::AXIsolatedObject::associatedAXObject const):
(WebCore::AXIsolatedObject::setMathscripts):
(WebCore::AXIsolatedObject::setObjectProperty):
(WebCore::AXIsolatedObject::setObjectVectorProperty):
(WebCore::AXIsolatedObject::setProperty):
(WebCore::AXIsolatedObject::insertText):
(WebCore::AXIsolatedObject::press):
(WebCore::AXIsolatedObject::initializeProperties): Deleted.
(WebCore::AXIsolatedObject::canBeMultilineTextField): Deleted.
(WebCore::AXIsolatedObject::setOptionalProperty): Deleted.
(WebCore::AXIsolatedObject::scrollView const): Deleted.
(WebCore::AXIsolatedObject::headingTagLevel const): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::AXIsolatedTree):
(WebCore::AXIsolatedTree::createEmptyContent):
(WebCore::AXIsolatedTree::create):
(WebCore::AXIsolatedTree::applyPendingRootNodeLocked):
(WebCore::AXIsolatedTree::storeTree):
(WebCore::AXIsolatedTree::nodeChangeForObject):
(WebCore::AXIsolatedTree::queueChange):
(WebCore::AXIsolatedTree::addUnconnectedNode):
(WebCore::AXIsolatedTree::resolveAppends):
(WebCore::AXIsolatedTree::queueAppendsAndRemovals):
(WebCore::AXIsolatedTree::collectNodeChangesForSubtree):
(WebCore::AXIsolatedTree::updateNode):
(WebCore::AXIsolatedTree::updateChildren):
(WebCore::AXIsolatedTree::setPendingRootNodeID):
(WebCore::AXIsolatedTree::setPendingRootNodeIDLocked):
(WebCore::AXIsolatedTree::willBeDestroyed):
(WebCore::AXIsolatedTree::applyPendingChanges):
(WebCore::AXIsolatedTree::processQueuedNodeUpdates):
(WebCore::convertToPropertyFlag):
(WebCore::setPropertyIn):
(WebCore::shouldCacheElementName):
(WebCore::canBeMultilineTextField):
(WebCore::createIsolatedObjectData):
(WebCore::AXIsolatedTree::applyPendingRootNode): Deleted.
(WebCore::AXIsolatedTree::setPendingRootNodeLocked): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.h:
(WebCore::IsolatedObjectData::IsolatedObjectData):
(WebCore::IsolatedObjectData::setProperty):
(WebCore::AXIsolatedTree::axObjectCacheID const):
(WebCore::AXIsolatedTree::rootNode):
(WebCore::AXIsolatedTree::NodeChange::NodeChange):
* Source/WebCore/accessibility/isolatedtree/mac/AXIsolatedObjectMac.mm:
(WebCore::appendBasePlatformProperties):
(WebCore::appendPlatformProperties):
(WebCore::AXIsolatedObject::initializeBasePlatformProperties): Deleted.
(WebCore::AXIsolatedObject::initializePlatformProperties): Deleted.
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase attachIsolatedObject:]):
(-[WebAccessibilityObjectWrapperBase hasIsolatedObject]):
(-[WebAccessibilityObjectWrapperBase detachIsolatedObject:]):
(-[WebAccessibilityObjectWrapperBase updateObjectBackingStore]):
(-[WebAccessibilityObjectWrapperBase axBackingObject]):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper _accessibilityShowContextMenu]):
(-[WebAccessibilityObjectWrapper _accessibilityChildrenFromIndex:maxCount:returnPlatformElements:]):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyFrameLoaderClient::setIsolatedTree):
(WebCore::EmptyFrameLoaderClient::setAXIsolatedTreeRoot): Deleted.
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::setIsolatedTree):
(WebKit::WebLocalFrameLoaderClient::setAXIsolatedTreeRoot): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectBase.h:
* Source/WebKit/WebProcess/WebPage/mac/WKAccessibilityWebPageObjectBase.mm:
(-[WKAccessibilityWebPageObjectBase accessibilityRootObjectWrapper:]):
(-[WKAccessibilityWebPageObjectBase setIsolatedTree:]):
(-[WKAccessibilityWebPageObjectBase setIsolatedTreeRoot:]): Deleted.
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::setIsolatedTree):
(WebKit::WebPage::setAXIsolatedTreeRoot): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.h:

Canonical link: https://commits.webkit.org/295174@main
